package/MDAnalysis/analysis/encore/utils.py

```diff
@@ -0,0 +1,444 @@
+        else:
+            self.progress = 100
+        return self
+
+    def __str__(self):
+        progressed = int(self.progress / self.step)  # fix
+        fill = progressed * self.fill
+        blank = (self.width - progressed) * self.blank
+        return self.format % {'fill': fill, 'blank': blank,
+                              'progress': int(self.progress)}
+
+    __repr__ = __str__
+
+    def _get_progress(self, increment):
+        return float(increment * 100) / self.end
+
+    def reset(self):
+        """Resets the current progress to the start point"""
+        self.progress = self._get_progress(self.start)
+        return self
+
+    def update(self, progress):
+        """Update the progress value instead of incrementing it"""
+        this_progress = self._get_progress(progress)
+        if this_progress < 100:
+            self.progress = this_progress
+        else:
+            self.progress = 100
+
+
+class AnimatedProgressBar(ProgressBar):
+    """Extends ProgressBar to allow you to use it straighforward on a script.
+    Accepts an extra keyword argument named `stdout`
+    (by default use sys.stdout).
+    The progress status may be send to any file-object.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(AnimatedProgressBar, self).__init__(*args, **kwargs)
+        self.stdout = kwargs.get('stdout', sys.stdout)
+
+    def show_progress(self):
+        if hasattr(self.stdout, 'isatty') and self.stdout.isatty():
+            self.stdout.write('\r')
+        else:
+            self.stdout.write('\n')
+        self.stdout.write(str(self))
+        self.stdout.flush()
+
+
+def trm_indeces(a, b):
+    """
+    Generate (i,j) indeces of a triangular matrix, between elements a and b.
+    The matrix size is automatically determined from the number of elements.
+    For instance: trm_indeces((0,0),(2,1)) yields (0,0) (1,0) (1,1) (2,0)
+    (2,1).
+
+    Parameters
+    ----------
+
+    `a` : (int i, int j) tuple
+        starting matrix element.
+
+    `b` : (int i, int j) tuple
+        final matrix element.
+    """
+    i, j = a
+    while i < b[0]:
+        if i == j:
+            yield (i, j)
+            j = 0
+            i += 1
+        else:
+            yield (i, j)
+            j += 1
+    while j <= b[1]:
+        yield (i, j)
+        j += 1
+
+
+def trm_indices_nodiag(n):
+    """generate (i,j) indeces of a triangular matrix of n rows (or columns),
+    without diagonal (e.g. no elements (0,0),(1,1),...,(n,n))
+
+    Parameters
+    ----------
+
+    `n` : int
+        Matrix size
+"""
+
+    for i in xrange(1, n):
+        for j in xrange(i):
+            yield (i, j)
+
+
+def trm_indices_diag(n):
+    """generate (i,j) indeces of a triangular matrix of n rows (or columns),
+    with diagonal
+
+    Parameters
+    ----------
+
+    `n` : int
+        Matrix size
+"""
+
+    for i in xrange(0, n):
+        for j in xrange(i+1):
+            yield (i, j)
+
+
+def merge_universes(universes):
```

Comment:
Is this concatenating the trajectories of the Universes?  Could do with more docs on how it's combining different Universes